templates/default.nginx_http.conf.template

```diff
@@ -15,6 +15,12 @@ log_format  custom  '$http_x_forwarded_for - $remote_user [$time_local] "$reques
 # workaround for the way we're including conf https://serverfault.com/questions/729128/overriding-nginx-access-log-directive-duplicate-log-entries
 access_log  off;
 
+# Don't send nginx version number
+server_tokens off;
+
+# Don't allow page to be rendered inside a frame or iframe
+add_header X-Frame-Options SAMEORIGIN;
```

Comment:
Why SAMEORIGIN and not DENY? May be out of date, but they recommend DENY here: https://cheatsheetseries.owasp.org/cheatsheets/REST_Security_Cheat_Sheet.html#Security_headers

Just asking, SAMEORIGIN may be OK.

Reply:
Hey Charles, thanks for pointing this out! I added `SAMEORIGIN` to stay more general, in case we are using iframes for anything, but `DENY` would be better if we are not.

Reply:
Was going to say probably not and then 
https://github.com/dockstore/dockstore-ui2/blob/ce2864d6a2ce3c3111326a40f9189ba3a0c25539/src/app/home-page/home-logged-out/home.component.ts#L35

But that probably doesn't count because we're serving content from somewhere else

Reply:
@denis-yuen You are correct, it doesn't count because the `SAMEORIGIN` would be for serving a Dockstore page using an iframe on another Dockstore page. I don't see any other use of frames or iframes in the dockstore-ui2 repo, so I will update the PR to use `DENY` instead of `SAMEORIGIN`.